### ISSUE TITLE
Wrong fallback max index in SumSpectra

### ISSUE THREAD

**Issue by SimonHeybrock** [mantidproject/mantid#15414]
`SumSpectra.cpp` line 101:

``` cpp
  // Something for m_maxSpec was given but it is out of range?
  if (!isEmpty(m_maxSpec) &&
      (m_maxSpec > m_numberOfSpectra - 1 || m_maxSpec < m_minSpec)) {
    g_log.warning("EndWorkspaceIndex out of range! Set to max Workspace Index");
    m_maxSpec = m_numberOfSpectra;
  }
```

Shouldn't the last lin read `m_maxSpec = m_numberOfSpectra - 1;`?

**Comment by SimonHeybrock** [comment]
Note that this might implicitly be fixed after we implement #15409.

**Comment by SimonHeybrock** [comment]
Same file, line 89:
```cpp
  // Check 'StartSpectrum' is in range 0-m_numberOfSpectra
  if (m_minSpec > m_numberOfSpectra) {
    g_log.warning("StartWorkspaceIndex out of range! Set to 0.");
    m_minSpec = 0;
  }
```

Should read `if (m_minSpec >= m_numberOfSpectra) {`, I guess?
